bridges/otellogr/logsink.go

```diff
@@ -238,9 +242,20 @@ func (l *LogSink) Info(level int, msg string, keysAndValues ...any) {
 	l.logger.Emit(ctx, record)
 }
 
+// WithContext returns a new LogSink with the specified context.
+func (l *LogSink) WithContext(ctx context.Context) *LogSink {
```

Comment:
Same as my previous comment, this adds a new feature, it's not needed to fix the panic issue. We should open it in another PR.